tests/test_patchy.py

```diff
@@ -185,6 +204,79 @@ def method(self):
 
         assert Artist().method() == "Crackers"
 
+    def test_patch_instancemethod_mangled(self):
+        class Artist(object):
+            def __mangled_name(self, v):
+                return v + ' on toast'
+
+            def method(self):
+                filling = 'Chalk'
+                return self.__mangled_name(filling)
+
+        patchy.patch(Artist.method, """\
+            @@ -1,2 +1,2 @@
+             def method(self):
+            -    filling = 'Chalk'
+            +    filling = 'Cheese'
+                 return self.__mangled_name(filling)
+            """)
+
+        assert Artist().method() == "Cheese on toast"
+
+    def test_patch_old_class_instancemethod_mangled(self):
+        class Artist:
+            def __mangled_name(self, v):
+                return v + ' on toast'
+
+            def method(self):
+                filling = 'Chalk'
+                return self.__mangled_name(filling)
+
+        patchy.patch(Artist.method, """\
+            @@ -1,2 +1,2 @@
+             def method(self):
+            -    filling = 'Chalk'
+            +    filling = 'Cheese'
+                 return self.__mangled_name(filling)
+            """)
+
+        assert Artist().method() == "Cheese on toast"
+
+    @pytest.mark.xfail
```

Comment:
why does this fail? because the new method is compiled without the original  `_Artist__mangled_name` in scope?